File: R/src/run_query.R
# Run this file inside an R console
source("R/src/query.R")

# Use this function to return a list of Taxon names that match your query
page_titles_vector <- ask_query_titles(query_string, output_file_name)
  
# Use this function to return a list of Taxon names and associated properties asked for by your query
properties_texts_data_frame <- ask_query_titles_properties(query_string, output_file_name)

File: python/src/run_query.py
# Run this file in your terminal like this: python -m src.run_query --output_file_name "file_name.csv" --query_string
# "[[Query::here]]"

import argparse
from src.query import *

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Run an Ask query against the FNA Beta API')
    parser.add_argument('--output_file_name', type=str, help='output_file_name')
    parser.add_argument('--query_string', type=str, help='query_string')
    args = parser.parse_args()
    ask_query(args.query_string, args.output_file_name)
